spec/lita/adapters/irc/cinch_plugin_spec.rb

```diff
@@ -98,4 +100,49 @@
       subject.on_room_join(m)
     end
   end
+
+  describe "#on_room_part" do
+    before do
+      allow(Lita::Room).to receive(:create_or_update).and_return(room)
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
+    end
+
+    it "triggers a part event when someone parts a channel" do
+      expect(robot).to receive(:trigger).with(
+        :user_parted_room,
+        user: user,
+        room: room,
+      )
+      subject.on_room_part(m)
+    end
+  end
+
+  describe "#on_quit" do
+    before do
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
+    end
+
+    it "triggers a disconnect event when someone quits from the IRC server" do
+      expect(robot).to receive(:trigger).with(
+        :user_disconnected,
+        user: user,
+      )
+      subject.on_quit(m)
+    end
+  end
+
+  describe "#on_nick_change" do
+    before do
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
```

Comment:
Maybe just change "from_user" to "old_user" and leave "user" as is. "from" is kinda ambiguous cause it sounds like it could mean "the username the person changed from" or "the user the event is from".